zlib.c

```diff
@@ -31,7 +31,7 @@ static const char *zerr_to_string(int status)
 #define ZLIB_BUF_MAX ((uInt) 1024 * 1024 * 1024) /* 1GB */
 static inline uInt zlib_buf_cap(size_t len)
 {
-	return (ZLIB_BUF_MAX < len) ? ZLIB_BUF_MAX : len;
+	return ((size_t) ZLIB_BUF_MAX < len) ? ZLIB_BUF_MAX : (uInt) len;
```

Comment:
I'm against implicit conversions for implementation defined concerns. One of them has to be probably wrong :wink: One of then is an up cast and the other a down cast, and they are being compared to each other...

Aside: for testing, 1.5Gb is better, as it crosses the thresholds quickly and not at boundaries.

Reply:
> I'm against implicit conversions for implementation defined concerns.

You can be against that all you want, at the same time you will have to abide by Git's conventions. And those conventions seem to avoid *quite hard* any unnecessary explicit cast.

For example, when `ZLIB_BUF_MAX` is compared to `len`, and they have different bit sizes, the smaller is up-cast to the larger one. That's defined behavior. Any compiler producing different code has a bug.

And do not forget that cluttering your code with unnecessary casts (or for that matter, cluttering your PR with unnecessary hunks) makes it much, much harder to read.

So no, this hunk needs to be dropped, sorry.